src/app/icd/ICDMonitoringTable.cpp

```diff
@@ -67,7 +90,72 @@ void ICDMonitoringEntry::Clear()
 {
     this->checkInNodeID    = kUndefinedNodeId;
     this->monitoredSubject = kUndefinedNodeId;
-    this->key              = ByteSpan();
+    if (symmetricKeystore != nullptr)
+    {
+        symmetricKeystore->DestroyKey(this->key);
+    }
+}
+
+CHIP_ERROR ICDMonitoringEntry::SetKey(ByteSpan keyData)
+{
+    VerifyOrReturnError(keyData.size() == sizeof(Crypto::Aes128KeyByteArray), CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrReturnError(symmetricKeystore != nullptr, CHIP_ERROR_INTERNAL);
+
+    // DeleteKey in case the handle was already used.
+    DeleteKey();
+
+    Crypto::Aes128KeyByteArray keyMaterial;
+    memcpy(keyMaterial, keyData.data(), sizeof(Crypto::Aes128KeyByteArray));
+
+    ReturnErrorOnFailure(symmetricKeystore->CreateKey(keyMaterial, key));
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR ICDMonitoringEntry::DeleteKey()
+{
+    VerifyOrReturnError(symmetricKeystore != nullptr, CHIP_ERROR_INTERNAL);
+    symmetricKeystore->DestroyKey(this->key);
+    return CHIP_NO_ERROR;
+}
+
+bool ICDMonitoringEntry::EnsureKeyEquivalent(ByteSpan keyData)
```

Comment:
Now that this is returning bool, `IsKeyEquivalent` would be a more sensible name.

Reply:
done